test/DependentAssemblyWithFunctions/Class1.cs

```diff
@@ -0,0 +1,31 @@
+﻿
+using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+
+namespace DependentAssemblyWithFunctions
+{
+    public class DependencyFunc
+    {
+        private readonly ILogger _logger;
+
+        public DependencyFunc(ILoggerFactory loggerFactory)
+        {
+            _logger = loggerFactory.CreateLogger<DependencyFunc>();
+        }
+
+        [Function("DependencyFunc")]
+        public HttpResponseData Run([HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestData req)
+        {
+            _logger.LogInformation("C# HTTP trigger function processed a request.");
+
+            var response = req.CreateResponse(HttpStatusCode.OK);
+            response.Headers.Add("Content-Type", "text/plain; charset=utf-8");
+
+            response.WriteString("Welcome to Azure Functions!");
+
+            return response;
+        }
```

Comment:
A function marked with `internal` visibility is ignored? Might be better to add that to this class and verify that in the test. 
Next time I have the same question, I can look at the tests to get an idea about the behavior.

Reply:
Also add a Static function.

Reply:
Tested legacy and functions marked with internal are NOT ignored. Same behavior with source-gen, test added.